src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,12 +1,50 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { Loader } from '../Loader';
+import { Todo } from '../../types/Todo';
+import { User } from '../../types/User';
+import { getUser } from '../../api';
+
+type Props = {
+  selectedTodo: Todo;
+  oncloseButtonClick: (value: number) => void;
+};
+
+export const TodoModal: React.FC<Props> = ({
+  selectedTodo,
+  oncloseButtonClick,
+}) => {
+  const [user, setUser] = useState<User | null>(null);
+  const [hasLoadingError, setHasLoadingError] = useState(false);
+
+  const hideModal = () => {
+    oncloseButtonClick(0);
+  };
+
+  const loadUserFromServer = async (currentTodo: Todo | null) => {
+    try {
+      if (currentTodo) {
+        const userFromServer = await getUser(currentTodo.userId);
+
+        setUser(userFromServer);
+        setHasLoadingError(false);
+      }
+    } catch {
+      setHasLoadingError(true);
+    }
+  };
+
+  useEffect(() => {
+    loadUserFromServer(selectedTodo);
+  }, []);
```

Comment:
please add missing dependency selectedTodo
```suggestion
  }, [selectedTodo]);
```